source/contentRecog/__init__.py

```diff
@@ -0,0 +1,259 @@
+		@param width: The width of the image in pixels.
+		@type width: int
+		@param height: The height of the image in pixels.
+		@type height: int
+		@param coordConverter: The converter to convert coordinates
+			in the supplied image to screen coordinates.
+			This should be used when returning coordinates to NVDA.
+		@type coordConverter: L{ResultCoordConverter}
+		@param onResult: A callable which takes a L{RecognitionResult} (or an exception on failure) as its only argument.
+		@type onResult: callable
+		"""
+		raise NotImplementedError
+
+	def cancel(self):
+		"""Cancel the recognition in progress (if any).
+		"""
+		raise NotImplementedError
+
+class ResultCoordConverter(object):
+	"""Converts coordinates in a recognition result to screen coordinates.
+	An image captured for recognition can begin at any point on the screen.
+	However, the image is cropped when passed to the recognizer.
+	Also, some recognizers need the image to be resized prior to recognition.
+	This converter converts coordinates in the recognized image
+	to screen coordinates suitable to be returned to NVDA;
+	e.g. in order to route the mouse.
+	"""
+
+	def __init__(self, left, top, resizeFactor):
+		"""
+		@param left: The x screen coordinate of the upper-left corner of the image.
+		@type left: int
+		@param top: The y screen coordinate of the upper-left corner of the image.
+		@type top: int
+		@param resizeFactor: The factor by which the image was resized for recognition.
+		@type resizeFactor: int or float
+		"""
+		self.left = left
+		self.top = top
+		self.resizeFactor = resizeFactor
+
+	def convertX(self, x):
+		"""Convert an x coordinate in the result to an x coordinate on the screen.
+		"""
+		return self.left + int(x / self.resizeFactor)
+
+	def convertY(self, y):
+		"""Convert an x coordinate in the result to an x coordinate on the screen.
+		"""
+		return self.top + int(y / self.resizeFactor)
+
+class RecognitionResult(object):
+	"""Provides access to the result of recognition by a recognizer.
+	The result is textual, but to facilitate navigation by word, line, etc.
+	and to allow for retrieval of screen coordinates within the text,
+	L{TextInfo} objects are used.
+	Callers use the L{makeTextInfo} method to create a L{TextInfo}.
+	Most implementers should use one of the subclasses provided in this module.
+	"""
+
+	def makeTextInfo(self, obj, position):
+		"""Make a TextInfo within the recognition result text at the requested position.
+		@param obj: The object to return for the C{obj} property of the TextInfo.
+			The TextInfo itself doesn't use this, but NVDA requires it to set the review object, etc.
+		@param position: The requested position; one of the C{textInfos.POSITION_*} constants.
+		@return: The TextInfo at the requested position in the result.
+		@rtype: L{textInfos.TextInfo}
+		"""
+		raise NotImplementedError
+
+# Used by LinesWordsResult.
+LwrWord = namedtuple("LwrWord", ("offset", "left", "top"))

```

Comment:
I'm a little confused by the name `LwrWord` does it stand for "Lines words Result Word"?